src/app/api/onboard/route.ts

```diff
@@ -1,31 +1,26 @@
-import { NextResponse } from "next/server";
-import { onboardingSchema, validateSchema } from "@/utils/schema";
+import { createValidationError } from "@/lib/schema/error";
+import { OnboardingRequest, OnboardingResponse } from "@/lib/schema/onboarding";
+import { validateSchema } from "@/lib/schema/validate";
 import { logger } from "@/utils/logger";
+import { NextResponse } from "next/server";
+import { z } from "zod";
 
 export async function POST(req: Request) {
-  try {
-    const body = await req.json();
-    const { validData, errors } = validateSchema(onboardingSchema, body);
+  const body = await req.json();
 
-    if (!validData || errors) {
-      logger.error("Validation failed");
-      return NextResponse.json(
-        { success: false, errors },
-        { status: 400 }
-      );
-    }
+  const { validData, errors } = validateSchema(OnboardingRequest, body);
+  if (!validData || errors) {
+    logger.debug("Validation failed", errors);
+    return NextResponse.json(createValidationError(errors), { status: 400 });
+  }
 
-    logger.info("Onboarding data received", validData);
+  logger.debug("Onboarding request", validData);
```

Comment:
@kunaldevxxx I used debug here because we need this during development only. Too many logs in production deployment causes issue with identifying actual errors that we care about